httpretty/core.py

```diff
@@ -318,15 +325,21 @@ def connect(self, address):
                 self.is_http = self._port in ports_to_check
 
             if not self.is_http:
-                if self.truesock:
+                if self.truesock and not self._connected_truesock:
                     self.truesock.connect(self._address)
+                    self._connected_truesock = True
                 else:
                     raise UnmockedError()
+            elif self.truesock is not None and not self._connected_truesock:
```

Comment:
Note: this change makes the `test_fakesock_socket_getpeercert` test from `test_core.py` actually connect out to somewhere.com, whereas I believe it did not do so before.